### PR TITLE
[search] Inverted a dependency.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
@ygorshenin PTAL

File: search/pre_ranker.hpp [hunk 1]
```diff
@@ -1,6 +1,12 @@
 #pragma once
 
+#include "indexer/index.hpp"
```
Comment (ygorshenin) [inline search/pre_ranker.hpp]: indexer should go after search

File: search/pre_ranker.hpp [hunk 2]
```diff
@@ -15,16 +21,39 @@ namespace search
...
+    // precision loss, its center may differ from
+    // |m_accuratePivotCenter|. Therefore the pivot should be used for
+    // fast filtering of features outside of the rectangle, while
+    // |m_accuratePivotCenter| should be used when it's needed to
+    // compute the distance from a feature to the pivot.
+    m2::PointD m_accuratePivotCenter = m2::PointD(0, 0);
+    int m_scale = 0;
+  };
+
+  explicit PreRanker(Index const & index, Ranker & ranker, size_t limit);
```
Comment (ygorshenin) [inline search/pre_ranker.hpp]: No need in ```explicit``` here.

File: search/pre_ranker.hpp [hunk 3]
```diff
@@ -15,16 +21,39 @@ namespace search
...
+
+  void Init(bool viewportSearch) { m_params.m_viewportSearch = viewportSearch; }
```
Comment (ygorshenin) [inline search/pre_ranker.hpp]: inline?
Also, please, insert a blank line after the method.

File: search/processor.cpp [hunk 4]
```diff
@@ -347,29 +345,38 @@ void Processor::Search(Results & results, size_t limit)
   Geocoder::Params geocoderParams;
   InitGeocoderParams(geocoderParams);
 
+  PreRanker::Params preRankerParams;
```
Comment (ygorshenin) [inline search/processor.cpp]: Where this variable is used?

File: search/processor.cpp [hunk 5]
```diff
@@ -347,29 +345,38 @@ void Processor::Search(Results & results, size_t limit)
...
+  InitPreRankerParams(preRankerParams);
+
   Ranker::Params rankerParams;
   InitRankerParams(rankerParams);
```
Comment (ygorshenin) [inline search/processor.cpp]: Where this variable is used?

File: search/processor.cpp [hunk 6]
```diff
@@ -591,10 +598,15 @@ void Processor::InitGeocoderParams(Geocoder::Params & params)
...
   params.m_mode = m_mode;
   params.m_pivot = GetPivotRect();
-  params.m_accuratePivotCenter = GetPivotPoint();
   m_geocoder.SetParams(params);
 }
 
+void Processor::InitPreRankerParams(PreRanker::Params & params)
+{
+  params.m_accuratePivotCenter = GetPivotPoint();
+  m_preRanker.SetParams(params);
```
Comment (ygorshenin) [inline search/processor.cpp]: The method name is disguising, as it not only inits PreRanker params, but also inits PreRanker with them.

Comment (ygorshenin) [inline search/processor.cpp]: I suggest you to get rid of m_preRanker.SetParams() call here. Or to rename the method, like:
```c++
void Processor::InitPreRanker();
```

File: search/ranker.hpp [hunk 7]
```diff
@@ -101,7 +87,7 @@ class Ranker
   void FlushViewportResults(Geocoder::Params const & geocoderParams, Results & res);
 
   void SetParams(Params const & params) { m_params = params; }
-
+  void SetPreResults1(vector<PreResult1> * preResults1) { m_preResults1 = preResults1; }
```
Comment (ygorshenin) [inline search/ranker.hpp]: You can completely get rid of pointers - just pass vector by rvalue reference and take the ownership here.

File: search/pre_ranker.hpp (hunk 1 above)
Comment (mpimenov) [inline search/pre_ranker.hpp]: done

File: search/pre_ranker.hpp (hunk 2 above)
Comment (mpimenov) [inline search/pre_ranker.hpp]: done

File: search/pre_ranker.hpp (hunk 3 above)
Comment (mpimenov) [inline search/pre_ranker.hpp]: done

File: search/processor.cpp (hunk 4 above)
Comment (mpimenov) [inline search/processor.cpp]: see below

File: search/processor.cpp (hunk 5 above)
Comment (mpimenov) [inline search/processor.cpp]: see below

File: search/processor.cpp (hunk 6 above)
Comment (mpimenov) [inline search/processor.cpp]: Removed the unused variables and changed to InitRanker/InitPreRanker.

File: search/ranker.hpp (hunk 7 above)
Comment (mpimenov) [inline search/ranker.hpp]: done

**Comment by mpimenov** [comment]
PTAL

File: search/processor.cpp [hunk 8]
```diff
@@ -347,29 +345,35 @@ void Processor::Search(Results & results, size_t limit)
...
 
-  InitParams(params);
-  params.m_mode = m_mode;
-  params.m_pivot = m_viewport[CURRENT_V];
-  params.m_accuratePivotCenter = params.m_pivot.Center();
-  m_geocoder.SetParams(params);
+  m_geocoder.SetParams(geocoderParams);
+
+  PreRanker::Params preRankerParams;
+  preRankerParams.m_accuratePivotCenter = geocoderParams.m_pivot.Center();
```
Comment (ygorshenin) [inline search/processor.cpp]: We need to call InitPreRanker() and InitRanker() here too, right?

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**